src/main/java/com/iota/iri/service/API.java

```diff
@@ -120,6 +122,23 @@ public API(Iota instance, IXI ixi) {
         maxRequestList = instance.configuration.integer(DefaultConfSettings.MAX_REQUESTS_LIST);
         maxGetTrytes = instance.configuration.integer(DefaultConfSettings.MAX_GET_TRYTES);
         maxBodyLength = instance.configuration.integer(DefaultConfSettings.MAX_BODY_LENGTH);
+
+        commandRoute = new HashMap<>();
+        commandRoute.put("addNeighbors", addNeighbors());
+        commandRoute.put("attachToTangle", attachToTangle());
+        commandRoute.put("broadcastTransactions", broadcastTransactions());
+        commandRoute.put("findTransactions", findTransactions());
+        commandRoute.put("getBalances", getBalances());
+        commandRoute.put("getInclusionStates", getInclusionStates());
+        commandRoute.put("getNeighbors", getNeighbors());
+        commandRoute.put("getNodeInfo", getNodeInfo());
+        commandRoute.put("getTips", getTips());
```

Comment:
why not just:
`commandRoute.put("getTips", getTipsStatement());`

and save yourself the unnecessary wrapper
`private Function<Map<String, Object>, AbstractResponse> getTips() {`

Reply:
I agree with memory problems in particular devices, specially with java applications. In this case I'm not sure if memory has increased with splitting this commands into functions but I'll check. Anyway I'll have this in mind for the future.